debezium-embedded/src/test/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotTest.java

```diff
@@ -1097,19 +1098,34 @@ public void testNotification() throws Exception {
 
         sendPauseSignal();
 
+        // Verify pause state in offset
+        final AtomicReference<SourceRecord> lastRecordBeforePause = new AtomicReference<>();
```

Comment:
WDYT instead of this to just create a real scenario tests. Send the signal, then pause, then restart the connector and then verify that the snapshot is still in pause state and then issue a resume.